### PR TITLE
GROOVY-9826: Better propagation of InterruptedException

### PR DESCRIPTION
**Pull request by paulk-asert**

### CODE REVIEW

**Comment by paulk-asert** [comment]
Proposed PR merged.

File: subprojects/groovy-jmx/src/test/java/groovy/jmx/builder/JmxConnectorHelper.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public static Map createRmiRegistry(int initPort) {
                 port = port + 1;
                 System.out.println("JmxBuilder - *** FAILED *** to create RMI Registry - Will Retry on port [" + port + "].");
                 try {
-                    Thread.currentThread().sleep(500);
+                    Thread.sleep(500);
                 } catch (InterruptedException e) {
                     throw new RuntimeException(e);
```
Comment (JLLeitschuh) [inline subprojects/groovy-jmx/src/test/java/groovy/jmx/builder/JmxConnectorHelper.java:50]: You still need to reset the interrupt flag here.

Comment (JLLeitschuh) [reply subprojects/groovy-jmx/src/test/java/groovy/jmx/builder/JmxConnectorHelper.java:50]: @paulk-asert 👋 was this ever resolved?

Comment (paulk-asert) [reply subprojects/groovy-jmx/src/test/java/groovy/jmx/builder/JmxConnectorHelper.java:50]: I created these:
https://issues.apache.org/jira/browse/GROOVY-10097
https://github.com/apache/groovy/pull/1579

**Pull request merged**